model_zoo/census_wide_deep_model/wide_deep_functional_api.py

```diff
@@ -0,0 +1,179 @@
+            logits=logits,
+        )
+    )
+
+
+def optimizer(lr=0.001):
+    return tf.keras.optimizers.Adam(learning_rate=lr)
+
+
+def eval_metrics_fn():
+    return {
+        "logits": {
+            "accuracy": lambda labels, predictions: tf.equal(
+                tf.cast(tf.reshape(predictions, [-1]) > 0.5, tf.int32),
+                tf.cast(tf.reshape(labels, [-1]), tf.int32),
+            )
+        },
+        "probs": {"auc": tf.keras.metrics.AUC()},
+    }
+
+
+def learning_rate_scheduler(model_version):
+    if model_version < 5000:
+        return 0.0003
+    elif model_version < 12000:
+        return 0.0002
+    else:
+        return 0.0001
+
+
+def dataset_fn(dataset, mode, metadata):
+    def _parse_data(record):
+        features = {}
+        for i, feature_name in enumerate(FEATURE_COLUMNS):
+            features[feature_name] = [record[i]]
+        label = tf.strings.to_number(record[-1], tf.int64)
+
+        return features, label
+
+    dataset = dataset.map(_parse_data)
+    return dataset
+
+
+def get_input_layers(feature_groups):
+    input_layers = {}
+    for feature_group in feature_groups.values():
+        for feature_info in feature_group:
+            input_layers[feature_info.name] = tf.keras.layers.Input(
+                name=feature_info.name, shape=(1,), dtype=feature_info.dtype
+            )
+    return input_layers
+
+
+def transform(inputs, feature_groups):
+    result = {}
+    for group_name, feature_group in feature_groups.items():
+        result[group_name] = transform_group(inputs, feature_group)
+    return result
+
+
+def transform_group(inputs, feature_group):
+    """Transform the inputs and concatenate inputs in a group
+    to a dense tensor
+    """
+    group_items = []
+    for feature_info in feature_group:
+        layer = get_transform_layer(feature_info)
+        transform_output = layer(inputs[feature_info.name])
+        group_items.append(transform_output)
+
+    id_offsets = get_id_boundaries(feature_group)
+
+    if id_offsets is not None:
+        group_items = AddIdOffset(id_offsets)(group_items)
+    group_stack = tf.keras.layers.concatenate(group_items, axis=-1)
+    return group_stack
+
+
+def get_transform_layer(feature_info):
+    if feature_info.op_name == TransformOp.LOOKUP:
+        return CategoryLookup(feature_info.param)
+    elif feature_info.op_name == TransformOp.HASH:
+        return CategoryHash(feature_info.param)
+    elif feature_info.op_name == TransformOp.BUCKETIZE:
+        return NumericBucket(feature_info.param)
+    else:
+        raise ValueError("The op is not supported")
+
+
+def get_id_group_dims(feature_groups):
+    id_group_dims = {}
+    for group_name, group_features in feature_groups.items():
+        id_boundaries = get_id_boundaries(group_features)
+        id_group_dims[group_name] = id_boundaries[-1]
+    return id_group_dims
+
+
+def wide_deep_model(input_layers, input_tensors, params):
```

Comment:
We can make `id_group_dims` and `embedding_config` as the specific parameter of the method `wide_deep_model`.
Putting them in the dict `params` is confusing.

Reply:
The `params` is a dict of hyperparameters like the args in Estimator.
https://www.tensorflow.org/api_docs/python/tf/estimator/Estimator#args